src/backend/distributed/utils/citus_stat_tenants.c

```diff
@@ -316,6 +321,8 @@ AttributeMetricsIfApplicable()
 		return;
 	}
 
+	QueryEndClock = clock();
```

Comment:
how do we ensure this only happens at the top level of a task?

e.g. if a query calls a function that function might separately invoke the executor, including ExecutorEnd_hook.

that's sort of ok for the purpose of counting, since this function does nothing if it is called multiple times for a single query, but we'd badly undercount CPU